include/sparrow/array.hpp

```diff
@@ -47,9 +55,21 @@ namespace sparrow
         SPARROW_API size_type size() const;
         SPARROW_API const_reference operator[](size_type) const;
 
+        SPARROW_API cloning_ptr<array_wrapper> && extract_array_wrapper() &&;
```

Comment:
The return type should be a value type instead of an rvalue reference to prevent dangling references.